Add stock quantity variable inside cart summary notification error

gadnis opened the pull request:
When user adjust quantity in cart summary and wanted quantity is more than 'in-stock', than error notification shown, but with this improvement user knows maximum product quantity he can buy.

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add maximum stock quantity with error notification.
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | [Forge](http://forge.prestashop.com/browse/BOOM-5987) ticket.
| How to test?  | If you know that for example Product quantity is 1, than in cart summary try to change quantity to 2 and buy it. You will get notification I'm talking about.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/9323)
<!-- Reviewable:end -->

gadnis committed:
Add stock quantity variable inside cart summary notification error

When user adjust quantity in cart summary and wanted quantity is more than 'in-stock', than error notification shown, but with this improvement user knows maximum product quantity he can buy.

gadnis committed:
Add stock quantity variable inside cart summary notification error

When user adjust quantity in cart summary and wanted quantity is more than 'in-stock', than error notification shown, but with this improvement user knows maximum product quantity he can buy.

gadnis commented:
How long does it usually take to approve a request ?

Quetzacoalt91 commented on controllers/front/CartController.php:
```diff
@@ -581,8 +581,8 @@ private function areProductsAvailable()
             return $this->trans(
-                'The item %product% in your cart is no longer available in this quantity. You cannot proceed with your order until the quantity is adjusted.',
-                array('%product%' => $product['name']),
+                'The item %product% in your cart is no longer available in this quantity. You cannot proceed with your order until the quantity is adjusted. Maximum quantity is %quantity%',
```
To be confirmed by the product team, but I guess the current message is used after checking the available quantity or, in this case, a product is enabled.

For instance, the maximum quantity of a product could be 20, but because it was disabled, you see this message. From a customer point of view, this is a non-sense.

LouiseBonnard replied:
+1

gadnis replied:
@Quetzacoalt91 I tried to disable product and the message "product is no longer available" is shown. So I believe it's working. Or maybe I misunderstood you?

Quetzacoalt91 replied:
What I meant is the maximum quantity is not needed here. A customer cannot order the product because it was disabled, not because he reached the maximum quantity.

gadnis replied:
You are right, thanks. I updated the file.

Edit: but wait.. now if I refresh page without changing quantity when product is enabled than I see old message without quantity variable. So after all we need that variable in the line I deleted..

gadnis committed:
Update CartController.php

gadnis committed:
Update CartController.php

marionf commented:
Hello @gadnis 

The quantity indicated for a product with combination is wrong:

![capture d ecran_654](https://user-images.githubusercontent.com/13449658/48940530-48892080-ef18-11e8-92d5-67f96ab69cba.png)

![capture d ecran_655](https://user-images.githubusercontent.com/13449658/48940532-4b841100-ef18-11e8-8715-bb06289d1858.png)

gadnis commented:
Somehow didn’t get an email about comment. Will try to check this issue asap

gadnis committed:
Fixed max quantity with attributes available

Now notification shows real quantity for both - products & products with attributes

gadnis reviewed (commented):
Hi, @marionf ,
it should be good now.

gadnis committed:
whatever

marionf commented:
@gadnis Thank you, I will retest it when the code review will be ok !

kpodemski commented:
I'm not sure if that's a good idea, not everyone wants to share informations about their stocks with customers

marionf commented:
@kpodemski 

This will be displayed only when there is more product in the cart than in stock.
This case rarely happens because normally the add to cart button on the product page is disabled when stock is <= 0 and order of out of stock product disabled.
It can help your customers to know how much they must decrement the quantity.

gadnis commented:
@kpodemski 

1. quantity is shown i product details in ps 1.7 by default. 

2. You can guess quantity when alert is shown after trying to increase quantity from 10 to 11. So why no to show that the maximum 10 products reach.

sarahdib commented:
Hello @gadnis 

Thank you for this contribution. In order to test this PR the conflicting files need to be resolved 

Have a nice day :)

PierreRambaud commented:
Hi @gadnis ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!
